### PR TITLE
#10840 Added doc-based support for project creation

### PR DESCRIPTION
**Pull request by daniel-jeremia**
* Added quickstart.doc.json as a example to create doc-based project

* Added DOCUMENT_BASED default extensions

### CODE REVIEW

File: quickstart/create/doc-based/document/doc-project-1/little prince.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+The Little Prince is a novella by French aristocrat, writer, and aviator Antoine de Saint-Exupéry.
```
Comment (jonathancesario) [inline quickstart/create/doc-based/document/doc-project-1/little prince.txt:1]: i prefer to use another lorem ipsum, perhaps name it `ipsum.txt`?

File: quickstart/create/doc-based/question-set.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+[
+  {
+    "type": "TEXT",
+    "config": {},
+    "name": "Q1",
+    "label": "Bound to Wind.Speed",
+    "bindToColumn": "Wind.Speed",
+    "required": true
```
Comment (ivanm-gdp) [inline quickstart/create/doc-based/question-set.json:8]: `bindToColumn` doesn't work for doc right?

maybe add a new question set file for doc based?

File: quickstart/config/quickstart.doc.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      "questionSetFile": "quickstart/create/doc-based/question-set.json",
+      "docFileOptions": {
+        "firstRowAsHeader": false,
+        "customHeaderColumns": [
+            {
+                "displayed": true,
+                "labelerRestricted": false,
+                "name": "Column 1"
+            }
+        ]
```
Comment (ivanm-gdp) [inline quickstart/config/quickstart.doc.json:58]: CMIIW, I think this does not affect anything - I tried creating the doc-based project with `"displayed": false` and the document still appear. 

copying from PCW, I got this
```json
"docFileOptions": {
  "customHeaderColumns": [],
  "firstRowAsHeader": false
}
```

but, running robosaur without `docFileOptions` also creates the project 🤔

**Comment by jonathancesario** [comment]
@ivanm-gdp any update?

**Comment by ivanm-gdp** [comment]
> @ivanm-gdp any update?

oh! sorry didn't realize this is waiting for my review
I have left a comment regarding [the question set](https://github.com/datasaur-ai/robosaur/pull/12#discussion_r828837809) - I don't think bind-to-column do anything document labeling right? 

If they don't, I think it should be better to have a separate example for doc-based

@jonathancesario @daniel-jeremia

**Comment by daniel-jeremia** [comment]
> 

got it, I'll use the example question set from gitbook

**Review by ivanm-gdp — APPROVED** [review]
Good to go from me 🚀 
Kak Rio's concern about new lines are already fixed too

**Comment by daniel-jeremia** [comment]
> Good to go from me rocket Kak Rio's concern about new lines are already fixed too

safe to merge ya? @ivanm-gdp @jonathancesario

**Pull request merged**